psydac/api/tests/navier_stockes.py

```diff
@@ -0,0 +1,280 @@
+# -*- coding: UTF-8 -*-
+
+import pytest
+import numpy as np
+from sympy import pi, cos, sin, sqrt, ImmutableDenseMatrix as Matrix, Tuple, lambdify
+from scipy.sparse.linalg import spsolve
+from scipy.sparse.linalg import gmres as sp_gmres
+from scipy.sparse.linalg import minres as sp_minres
+from scipy.sparse.linalg import cg as sp_cg
+from scipy.sparse.linalg import bicg as sp_bicg
+from scipy.sparse.linalg import bicgstab as sp_bicgstab
+
+from sympde.calculus import grad, dot, inner, div, curl, cross
+from sympde.topology import NormalVector
+from sympde.topology import ScalarFunctionSpace, VectorFunctionSpace
+from sympde.topology import ProductSpace
+from sympde.topology import element_of, elements_of
+from sympde.topology import Square, Union
+from sympde.expr     import BilinearForm, LinearForm, integral
+from sympde.expr     import Norm
+from sympde.expr     import find, EssentialBC
+
+from psydac.api.essential_bc   import apply_essential_bc
+from psydac.cad.geometry       import refine_knots
+from psydac.fem.basic          import FemField
+from psydac.fem.vector         import ProductFemSpace
+from psydac.api.discretization import discretize
+from psydac.linalg.utilities   import array_to_stencil
+from psydac.linalg.stencil     import *
+from psydac.linalg.block       import *
+from psydac.api.settings       import PSYDAC_BACKEND_GPYCCEL
+
+from psydac.linalg.iterative_solvers import cg, pcg, bicg
+
+from sympde.calculus import laplace, grad, Transpose
+from sympde.expr     import TerminalExpr
+
+from mpi4py import MPI
+comm = MPI.COMM_WORLD
+#==============================================================================
+def get_boundaries(*args):
+
+    if not args:
+        return ()
+    else:
+        assert all(1 <= a <= 4 for a in args)
+        assert len(set(args)) == len(args)
+
+    boundaries = {1: {'axis': 0, 'ext': -1},
+                  2: {'axis': 0, 'ext':  1},
+                  3: {'axis': 1, 'ext': -1},
+                  4: {'axis': 1, 'ext':  1}}
+
+    return tuple(boundaries[i] for i in args)
+#==============================================================================
+def run_navier_stokes_2d(domain, f, ue, pe, *, ncells, degree):
+
```

Comment:
You may specify here that this solves the steady-state problem

Reply:
done